coderd/database/migrations/000256_workspace_agent_script_timings.up.sql

```diff
@@ -0,0 +1,19 @@
+ALTER TABLE workspace_agent_scripts ADD COLUMN id           uuid UNIQUE NOT NULL DEFAULT gen_random_uuid();
+ALTER TABLE workspace_agent_scripts ADD COLUMN display_name text        NOT NULL DEFAULT '';
+
+CREATE TYPE workspace_agent_script_timing_stage AS ENUM (
+    'start',
+    'stop',
+    'cron'
+    );
+
+CREATE TABLE workspace_agent_script_timings
+(
+    script_id     uuid                                NOT NULL REFERENCES workspace_agent_scripts (id) ON DELETE CASCADE,
```

Comment:
It's a bit awkward to reference a unique timing since this table doesn't have an `id`. You need a combination of `(script_id, started_at[, ended_at])`, probably OK but worth a note.

There's no protection against inserting duplicates here though, unique on `(script_id, started_at)` is probably fine. (This will also enable upserts if you decide to implement my suggestion for sending two payloads, start and complete.)

Reply:
I'm happy to add an `id` field to this table if that makes our life easier.

Reply:
We can always add it after the fact, if we add the unique constraint we essentially have a unique ID so I think that's fine. 👍🏻